evalml/objectives/standard_metrics.py

```diff
@@ -345,6 +345,17 @@ def objective_function(self, y_true, y_predicted, X=None):
         return metrics.mean_absolute_error(y_true, y_predicted)
 
 
+class MAPE(RegressionObjective):
+    """Mean absolute percentage error for time series regression."""
+    name = "Mean Absolute Percentage Error"
+    greater_is_better = False
+    score_needs_proba = False
+    perfect_score = 0.0
+
+    def objective_function(self, y_true, y_predicted, X=None):
+        return (np.abs((y_true - y_predicted) / y_true))[y_true != 0].mean() * 100
```

Comment:
@ParthivNaresh nice, looks good!

Sklearn 0.24.2 (a current release candidate) [has a MAPE impl](https://github.com/scikit-learn/scikit-learn/blob/4773f3e39/sklearn/metrics/_regression.py#L197) ([docs](https://scikit-learn.org/dev/modules/generated/sklearn.metrics.mean_absolute_percentage_error.html)).

I like [their use of `max(y_true, eps)`](https://github.com/scikit-learn/scikit-learn/blob/4773f3e39/sklearn/metrics/_regression.py#L261) in the denominator. I think we should do that instead of excluding rows where `y_true` is exactly 0.

I also liked that they had weight support. We don't have a pattern for this yet from other objectives, and its not a requirement for this PR, but would be nice to have.

Reply:
@dsherry That's an interesting approach, but wouldn't that result in an extremely high value in the overall calculation? @freddyaboulton

Reply:
Good point @dsherry! sklearn's definition of `eps` is way too small but even if we used our own I'm worried that there'll always be one use case where that value is "wrong".

I prefer treating this how we treat RootMeanSquaredLogError or MeanSquaredLogError. Those objectives error out if passed targets with any negative values. They are also kept out of the core objectives because the user only uses them when they are defined for their particular dataset. Maybe we should add MAPE to `get_non_core_objectives`

![image](https://user-images.githubusercontent.com/41651716/101396772-d33bde80-3899-11eb-9ab4-020be932a2f1.png)


![image](https://user-images.githubusercontent.com/41651716/101396528-72aca180-3899-11eb-8359-35a7b388887f.png)

Reply:
> sklearn's definition of eps is way too small

Hmm, its an arbitrary threshold, right? Setting it to min numerical precision makes sense to me because any differences smaller than that (more or less) cannot be trusted.

It seems we have a few options for what to do when the target contains values equal to or super close to zero (options not all mutually exclusive):
1. Have the objective error out.
2. Have the objective round the zero value up to a small eps in the denominator.
3. Have automl error out during init if MAPE was selected as the automl objective. Similar treatment for RMSLE/MSLE if the target has negative values.

What behavior is most useful to users?

@freddyaboulton @ParthivNaresh I don't have a strong preference between 1 and 2--if you're both advocating for 1, let's do that in this PR. However to really nail that question I think we should also file 3 as an enhancement and address it separately.

Reply:
I think the problem with the sklearn approach is that it will divide by eps whenever y=0 which will cause the score to blow up.

![image](https://user-images.githubusercontent.com/41651716/101415999-e7da9f80-38b6-11eb-9494-5d93abeb4446.png)

I don't think scores that large are useful to users, especially if they have a dataset where y=0 occurs frequently. I like option 1 because I prefer letting the user know at the beginning that this objective isn't great for problems with y=0 (rather than having them comb through or impl trying to reason why their objective is ~1e14). 

You bring up a good point about AutoMLSearch! We should definite file that. Currently we never allow RMSLE and MSLE as the primary objective but I like the idea of only disallowing it if the target has negative values. Similar for MAPE. That sounds like it could be a data check!

Reply:
@freddyaboulton agreed! And ooh yeah definitely, perhaps that could be part of the "InvalidTargetsDataCheck"

Reply:
Filed #1521 !